src/register.rs

```diff
@@ -23,134 +23,237 @@ pub fn register(filename: &String, option: &String) -> Result<(), std::io::Error
         .filter(|x| match option.as_str() {
             "all" => true,
             _ => {
-                x.acct_type.contains(option)
-                    || x.date.contains(option)
-                    || x.acct_name.contains(option)
-                    || x.acct_offset_name.contains(option)
-                    || x.name.contains(option)
-                    || x.debit_credit.to_string().contains(option)
+                let optional_account = match &x.account {
+                    None => "optional:account".to_string(),
+                    Some(name) => name.to_string(),
```

Comment:
Probably good idea to add this match option. The account passed in the filter likely won't have any trailing spaces but what is stored in the text file may.

```suggestion
                    Some(name.trim()) => name.trim().to_string(),
```